Events

AdnanAbuElNadi opened the pull request:
Responsive Events Section done with react-slick

abukhadraop commented on src/components/common/EventCardContainer/event-card-container.styles.js:
```diff
@@ -0,0 +1,67 @@
+    display: none;
+  }
+
+  @media (min-width: 22.5rem) {
```
there is no need to specify the width with all media queries, we can let the components to grow and shrink to fill the container element, and you can cheng the slider settings to view the number of items you want on each screen (i think 4 items in desktop 3 or 2 items in tablet and 1 item for mobile is enough )

AdnanAbuElNadi replied:
Im trying to use the exact measurements in the figma design , in a 1440p screen width they had 4 cards with 16px(1rem) gap between them  and a 56px (3.5rem) margin , and a 20px (1.25rem) on the mobile screen, using these and the fixed width of the cards i had to set a slider width for every screen because some of the cards do not overflow completely,
furthermore , not setting a width to the slider makes it infinite and the cards go off screen

abukhadraop reviewed (changes requested):
General notes 
- make sure that your code is well documented 
- please pay attention to the console log errors and fix them
- please clean up the unneeded code, we should not push any commented code to GH
- never ever use inline styles, all elements styles should be encapsulated in styled components
- do not write any media queries in the styles we have an utility that generates media queries   based on unified breaking points, look at `src/styles/breakpoints.js` and  `src/styles/media-query.js`
   you can use it by something like this
   ```js
   import mq from 'styles/media-query';

   ....

   ${mq.tablet`
     style:value;
   `}
   ```
- we should keep an empty line between CSS blocks 
   ```js
    const Item = styled.div`
      ...
      // empty line
      p {
       ...
      }
     // empty line
      ${mq.tablet`
       ....
      `}
    `;
   ``` 
- look like the component does not behave correctly when i change the  event type
<img width="1792" alt="Screen Shot 2021-12-12 at 5 31 55 PM" src="https://user-images.githubusercontent.com/53254594/145719294-c14b9fb5-4340-4ff5-8a4d-b1c64b2f8d15.png">


<img width="1792" alt="Screen Shot 2021-12-12 at 5 31 58 PM" src="https://user-images.githubusercontent.com/53254594/145719300-8ea20092-489f-4cec-96f8-73570f7da166.png">

AdnanAbuElNadi commented:
Added some documentation.
Removed any console logs
Removed All commented Code
Removed inline styles
Fixed media queries
Fixed Style Blocks spacing
Fixed the Errors with the slider and filter
Removed Global Styled From index.js
Removed any extra styling from the global-styles.js
Fixed colors casing and Removed Duplicates

Firgma Mobile Events Section : https://www.figma.com/file/22d3bKcFL0ipWIZJOUY3Km/reverifi?node-id=0%3A16338
Firgam Desktop Events Section : https://www.figma.com/file/22d3bKcFL0ipWIZJOUY3Km/reverifi?node-id=0%3A3779

AdnanAbuElNadi commented:
Changed useEffect for a function that outputs a filtered Events Array as a prop to the slider
Added Missing Documentation
Finished Testing with 100% coverage

abukhadraop commented:
not needed
